### DISCUSSION TITLE
Multitenancy: Project Scoping?

### DISCUSSION THREAD

**Discussion by chantlong** [directus/directus#16559] [Archive]
Hi there,
I'm trying to do multitenancy with Directus where different tenants share a single database.
Role scoping will probably work for me but I'm trying to understand Project Scoping.

> Project Scoping — Creating a super-admin layer that provisions new tenant projects has been made easier by the Cloud-native model of Directus 9+. This method involves developing custom code that can dynamically spin up/down projects, but is also the most flexible, supporting scoped extensions and differentiated project settings.

How do you create multiple Projects through the UI?  Can multiple projects share the same single database or each project requires its own database?
Is there sample of this custom code?

Thanks

**Comment by br41nslug** [comment]
> This method involves developing custom code that can dynamically spin up/down projects

Meaning you're creating multiple instances of Directus (one per client) like the cloud does. So this cannot be done from the UI as it is higher over than one single Directus project. For instance with dockers you'd need custom code to create/stop/monitor the individual containers you're using for clients. With this approach you'd generally not share the database while thats possible it sort of defeats the purpose of choosing this approach (or they could all use the same database server but different databases)

**Reply by MentalGear** [reply]
@chantlong  might refer to a scenario where you share the same db structure, but not the same db data. E.g. tenants are different groups that do not see each other's data. A bit like what I'm discussing: https://github.com/directus/directus/discussions/16533
